log/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+use lv2_core::feature::{Feature, ThreadingClass};
+//use std::ffi::CString;
+use std::os::raw::*; //get all common c_type
+use urid::*;
+
+pub struct EntryClass;
+unsafe impl UriBound for EntryClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Entry;
+}
+
+/// UriBound for an error message.
+pub struct ErrorClass;
+unsafe impl UriBound for ErrorClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Error;
+}
+
+/// UriBound for an informative message.
+pub struct NoteClass;
+unsafe impl UriBound for NoteClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Note;
+}
+
+/// UriBound for a debuging message.
+pub struct TraceClass;
+unsafe impl UriBound for TraceClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Trace;
+}
+
+/// Uribound for an error message.
+pub struct WarningClass;
+unsafe impl UriBound for WarningClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Warning;
+}
+
+/// Marker for URID representing the nature of a log message.
+// Note : it's may be better to have a URID trait to define a common interface
+pub unsafe trait EntryType
+where
+    Self: Copy,
+{
+}
+
+unsafe impl EntryType for URID<ErrorClass> {}
+unsafe impl EntryType for URID<NoteClass> {}
+unsafe impl EntryType for URID<TraceClass> {}
+unsafe impl EntryType for URID<WarningClass> {}
+
+/// Errors potentially generated by [`Log`](struct.Log.html) methods.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum LogError {
```

Comment:
The LV2 Spec doesn't define what error types can be called yet, so this probably should be an empty struct instead:

```rust
#[derive(PartialEq, Eq, Clone, Copy, Debug)]
pub struct LogError {
    _private: ()
}
```

(the private field is here in case error codes are added in the future).

Also, because this is an error type, this must also implement the [`Error`](https://doc.rust-lang.org/std/error/trait.Error.html) trait (see the Rust API Guidelines: https://rust-lang.github.io/api-guidelines/interoperability.html#error-types-are-meaningful-and-well-behaved-c-good-err )

Reply:
Lol, i think you didn't understood what was the intention here. I did that because previous implementation could fail for different reason. Now, only printing error can happen, so i replaced LogError by an empty PrintError struct and I implemented the Error trait on it.